### PR TITLE
feat: Add Mainnet deployments for Beacon (L2<>L1) contracts + new hardhat features

### PR DESCRIPTION
**Pull request by nicholaspai**
**Motivation**

- Add ability for `hardhat-deploy` to read deployed contract data from `core/networks`
- Add a task to setup `Bridge`.

**Testing**

Check a box to describe how you tested these changes and list the steps for reviewers to test.

- [ ]  Ran end-to-end test, running the code as in production
- [ ]  New unit tests created
- [ ]  Existing tests adequate, no new tests required
- [ ]  All existing tests pass
- [X]  Untested

### CODE REVIEW

File: packages/common/src/HardhatConfig.js [hunk 1]
```diff
@@ -1,4 +1,6 @@
 const { getNodeUrl, mnemonic } = require("./TruffleConfig");
+const path = require("path");
+const coreWkdir = path.dirname(require.resolve("@uma/core/package.json"));
```
Comment (nicholaspai) [inline packages/common/src/HardhatConfig.js:3]: @mrice32 did you figure out a better way to get this path without having to import `@uma/core`?

Comment (mrice32) [reply packages/common/src/HardhatConfig.js:3]: We can do it the same way we do it in TruffleConfig. The caller can pass `workingDirectory` into `getHardhatConfig`.

Thoughts?

**Review by mrice32 — APPROVED** [review]
LGTM!

File: packages/common/src/HardhatConfig.js [hunk 2]
```diff
@@ -1,8 +1,7 @@
 const { getNodeUrl, mnemonic } = require("./TruffleConfig");
 const path = require("path");
-const coreWkdir = path.dirname(require.resolve("@uma/core/package.json"));
 
-function getHardhatConfig(configOverrides) {
+function getHardhatConfig(configOverrides, truffleContextDir = "./") {
```
Comment (mrice32) [inline packages/common/src/HardhatConfig.js:4]: nit: why not name this `workingDir`? Doesn't seem to have much to do with truffle right?

**Review by mrice32 — APPROVED** [review]
LGTM!

File: packages/common/src/HardhatConfig.js (hunk 1 above)
Comment (nicholaspai) [reply packages/common/src/HardhatConfig.js]: Done

**Pull request merged**